ThatDefaultProject/Assets/ThatDefaultProject/Scripts/SpawningUtils.cs

```diff
@@ -0,0 +1,129 @@
+using UnityEngine;
+
+namespace that
+{
+    public static class SpawningUtils
+    {
+		// https://discussions.unity.com/t/instantiate-inactive-object/39830/3
+
+		/// <summary>
+		/// Will instantiate an object disabled preventing it from calling Awake/OnEnable.
+		/// </summary>
+		public static T InstantiateDisabled<T>(this T original, Transform parent = null, bool worldPositionStays = false) where T : Object
+		{
+			if (!GetActiveState(original))
+			{
+				return Object.Instantiate(original, parent, worldPositionStays);
+			}
+
+			(GameObject coreObject, Transform coreObjectTransform) = CreateDisabledCoreObject(parent);
+			T instance = Object.Instantiate(original, coreObjectTransform, worldPositionStays);
+			SetActiveState(instance, false);
+			SetParent(instance, parent, worldPositionStays);
+			Object.Destroy(coreObject);
+			return instance;
+		}
+
+		/// <summary>
+		/// Will instantiate an object disabled preventing it from calling Awake/OnEnable.
+		/// </summary>
+		public static T InstantiateDisabled<T>(this T original, Vector3 position, Quaternion rotation, Transform parent = null) where T : Object
+		{
+			if (!GetActiveState(original))
+			{
+				return Object.Instantiate(original, position, rotation, parent);
+			}
+
+			(GameObject coreObject, Transform coreObjectTransform) = CreateDisabledCoreObject(parent);
+			T instance = Object.Instantiate(original, position, rotation, coreObjectTransform);
+			SetActiveState(instance, false);
+			SetParent(instance, parent, false);
+			Object.Destroy(coreObject);
+			return instance;
+		}
+
+		private static (GameObject coreObject, Transform coreObjectTransform) CreateDisabledCoreObject(Transform parent = null)
```

Comment:
Wouldn't you also call this Instantiate? so InstantiateDisabledCoreObject
I also think RootObject is a better name, but that's just personal taste

Reply:
Fixed in ec3151440d9f560d2bbfa68b2195ee64841c0ccb